src/enclave/Enclave/Crypto.cpp

```diff
@@ -1,45 +1,31 @@
-#include "Crypto.h"
-
 #include <stdexcept>
-#include <sgx_trts.h>
-#include <sgx_tkey_exchange.h>
 
 #include "common.h"
+#include "Crypto.h"
+#include "Random.h"
 #include "util.h"
 
+
 /**
  * Symmetric key used to encrypt row data. This key is shared among the driver and all enclaves.
  *
  * The key is initially set on the driver, as the Scala byte array
  * edu.berkeley.cs.rise.opaque.Utils.sharedKey. It is securely sent to the enclaves if attestation
  * succeeds.
  */
-sgx_aes_gcm_128bit_key_t shared_key = {0};
+unsigned char shared_key[SGX_AESGCM_KEY_SIZE] = {0};
 
 std::unique_ptr<KeySchedule> ks;
 
 void initKeySchedule() {
   ks.reset(new KeySchedule(reinterpret_cast<unsigned char *>(shared_key), SGX_AESGCM_KEY_SIZE));
 }
 
-void set_shared_key(sgx_ra_context_t context, uint8_t *msg4_bytes, uint32_t msg4_size) {
-  if (msg4_size != sizeof(ra_msg4_t)) {
+void set_shared_key(uint8_t *shared_key_bytes, uint32_t shared_key_size) {
+  if (shared_key_size <= 0) {
     throw std::runtime_error("Remote attestation step 4: Invalid message size.");
```

Comment:
Modify the runtime error here? Perhaps to something like "Invalid symmetric key passed to enclave`. We should scrap referral to attestation steps.